### PR TITLE
Add `new_unchecked` method for `AffinePoint`

### PR DESCRIPTION
**Pull request by tcoratger**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the new behavior?

- Add `new_unchecked` method for `AffinePoint`

## Does this introduce a breaking change?

No

### CODE REVIEW

File: crates/starknet-types-core/src/curve/affine_point.rs [hunk 1]
```diff
@@ -18,6 +18,14 @@ impl AffinePoint {
         )?))
     }
 
+    pub fn new_unchecked(x: Felt, y: Felt) -> AffinePoint {
```
Comment (LucasLvy) [inline crates/starknet-types-core/src/curve/affine_point.rs:21]: Can we make this function const ?

Comment (tcoratger) [reply crates/starknet-types-core/src/curve/affine_point.rs]: Yes sure, done

File: crates/starknet-types-core/src/curve/affine_point.rs [hunk 2]
```diff
@@ -18,6 +18,14 @@ impl AffinePoint {
         )?))
     }
 
+    pub const fn new_unchecked(x: Felt, y: Felt) -> AffinePoint {
```
Comment (LucasLvy) [inline crates/starknet-types-core/src/curve/affine_point.rs:21]: What’s the behavior if you send incorrect values ? Does it panic ? Can you add a small description over the function explaining that plz

Comment (tcoratger) [reply crates/starknet-types-core/src/curve/affine_point.rs:24]: Just updated the documentation here. 

As per what is done here in the same context for Arkworks https://github.com/arkworks-rs/algebra/blob/bb663bc27e776af81e66c94d9016bde2dbba9968/ec/src/models/short_weierstrass/group.rs#L120, this method does not check if the point is on the curve or not.

As its name suggests, the method must be used with caution and full knowledge of the facts for very specific use cases.

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**